packages/@ionic/cli/src/definitions.ts

```diff
@@ -576,6 +586,14 @@ export interface VueBuildOptions extends BuildOptions<'vue'> {
   cordovaAssets?: boolean;
 }
 
+export interface IonicCapacitorOptions {
+  configuration?: string;
```

Comment:
`configuration` is specific to Angular. We should reuse the hook input from build/serve, which is the same as the runner options. https://github.com/ionic-team/ionic-cli/blob/f4f05ffe4649a66f4c7961ee09bea97343d0e4f3/packages/%40ionic/cli/src/lib/build.ts#L93-L95

https://github.com/ionic-team/ionic-cli/blob/f4f05ffe4649a66f4c7961ee09bea97343d0e4f3/packages/@ionic/cli/src/commands/capacitor/base.ts#L102

Reply:
If we do that, then we'll rule out adding other capacitor-specific information in  later update, like the `appId` for instance.

Reply:
I was thinking `CapacitorBuildHookInput` would have a `build` key much like `BuildHookInput` and then have an additional key of Capacitor-specific info. `CapacitorRunHookInput` would have either a `build` or a `serve` key, depending on if `--livereload` is used or not. Does that make sense?

Reply:
That sounds great, in that way it is even more flexible.

Reply:
I have updated the pull request, the hook input now looks like this:

``` typescript
export interface CapacitorRunHookInput {
  readonly name: 'capacitor:run';
  readonly serve?: AngularServeOptions | IonicAngularServeOptions | Ionic1ServeOptions;
  readonly build?: AngularBuildOptions | IonicAngularBuildOptions | Ionic1BuildOptions;
  readonly capacitor: IonicCapacitorOptions;
}

export interface CapacitorBuildHookInput {
  readonly name: 'capacitor:build';
  readonly build: AngularBuildOptions | IonicAngularBuildOptions | Ionic1BuildOptions;
  readonly capacitor: IonicCapacitorOptions;
}

export interface IonicCapacitorOptions {
  verbose?: boolean;
  appId?: string;
  appName?: string;
  server?: {
    url?: string;
  };
  '--': string[];
}
```